plugins/youtube.py

```diff
@@ -7,17 +7,22 @@
 from pytube.exceptions import PytubeError
 from yt_dlp import YoutubeDL
 
-from core.options import Options
+from core.options import Options, Choice
 from ytdl_config import ytdl_opts
 from core.model import PodcastItem, PodcastFeed
 from core.exceptions import PluginError, InputError
 from core.plugin.plugin import Plugin
 
 
+class YoutubeLibrary(Choice):
+    pytube = "pytube"
+    yt_dlp = "yt-dlp"
+
+
 class PluginImpl(Plugin):
 
     class PluginOptions(Options):
-        library = 'pytube'
+        library: YoutubeLibrary = 'yt-dlp'
```

Comment:
Currently broken: https://github.com/pytube/pytube/issues/1299. Seems like youtube-dl is better maintained anyways.